src/app.exec.ts

```diff
@@ -166,6 +180,43 @@ async function exec(ass: RV1, pkgs: PackageSpecification[], opts: {env: boolean}
 
 ////
 
+import {readLines} from "deno/io/buffer.ts"
+
+async function extract_shebang(path: Path) {
+  const f = await Deno.open(path.string, { read: true })
+  const line = (await readLines(f).next()).value
+  const shebang = line.match(/^\s*#!(\S+)$/)?.[1]
+  if (shebang) {
+    return Path.abs(shebang)?.basename()
+  }
+}
+
+async function which(arg0: string) {
+  /// some special casing because we cannot represent version’d stuff in pantry yet
+  switch (arg0) {
+  case 'python2':
+    return { project: 'python.org', constraint: new semver.Range("2") }
+  case 'python3':
+    return { project: 'python.org', constraint: new semver.Range("3") }
+  }
+
+  const pantry = usePantry()
+  let found: { project: string } | undefined
+
+  for await (const entry of pantry.ls()) {
+    if (found) break
+    pantry.getProvides(entry).then(provides => {
+      if (!found && provides.includes(arg0)) {
```

Comment:
If we strip the bin/ prefix, should we drop that in the pantry?

Reply:
maybe. I added it to pantry so in future we *could* eg. provide other UNIX dirs eg. `share/scripts/foo` might well be something, or ofc `sbin/foo`